src/msw/graphicsd2d.cpp

```diff
@@ -360,6 +360,221 @@ HRESULT WINAPI wxD3D11CreateDevice(
+            {
+                hr = m_factory->CreateFontFileReference(m_filePaths[m_nextIndex].wc_str(), NULL, &m_currentFile);
+                if ( SUCCEEDED(hr) )
+                {
+                    *pHasCurrentFile = TRUE;
+                    ++m_nextIndex;
+                }
+            }
+
+            return hr;
+        }
+
+        virtual HRESULT STDMETHODCALLTYPE GetCurrentFontFile(IDWriteFontFile** ppFontFile) wxOVERRIDE
+        {
+            if ( m_currentFile )
+            {
+                m_currentFile.get()->AddRef();
+            }
+            *ppFontFile = m_currentFile;
+
+            return (m_currentFile != NULL) ? S_OK : E_FAIL;
+        }
+
+    private:
+        ULONG m_refCount;
+
+        wxCOMPtr<IDWriteFactory> m_factory;
+        wxCOMPtr<IDWriteFontFile> m_currentFile;
+        wxArrayString m_filePaths;
+        size_t m_nextIndex;
+    };
+
+    class wxDirect2DFontCollectionLoader : public IDWriteFontCollectionLoader
+    {
+    public:
+        wxDirect2DFontCollectionLoader()
+            : m_refCount(0)
+        {
+        }
+
+        // IUnknown methods
+        virtual HRESULT STDMETHODCALLTYPE QueryInterface(REFIID iid, void** ppvObject) wxOVERRIDE
+        {
+            if ( IsEqualIID(iid, IID_IUnknown) || IsEqualIID(iid, __uuidof(IDWriteFontCollectionLoader)) )
+            {
+                *ppvObject = this;
+                AddRef();
+                return S_OK;
+            }
+            else
+            {
+                *ppvObject = NULL;
+                return E_NOINTERFACE;
+            }
+        }
+
+        virtual ULONG STDMETHODCALLTYPE AddRef() wxOVERRIDE
+        {
+            return ::InterlockedIncrement(&m_refCount);
+        }
+
+        virtual ULONG STDMETHODCALLTYPE Release() wxOVERRIDE
+        {
+            ULONG newCount = ::InterlockedDecrement(&m_refCount);
+            if ( newCount == 0 )
+                delete this;
+
+            return newCount;
+        }
+
+        // IDWriteFontCollectionLoader methods
+        virtual HRESULT STDMETHODCALLTYPE CreateEnumeratorFromKey(IDWriteFactory* pFactory,
+                                            void const* pCollectionKey, UINT32 collectionKeySize,
+                                            IDWriteFontFileEnumerator** pFontFileEnumerator) wxOVERRIDE
+        {
+            if ( !pFontFileEnumerator )
+                return E_INVALIDARG;
+
+            *pFontFileEnumerator = NULL;
+
+            if ( collectionKeySize != sizeof(wxDirect2DFontKey) )
+                return E_INVALIDARG;
+
+            wxDirect2DFontFileEnumerator* pEnumerator = new wxDirect2DFontFileEnumerator(pFactory);
+            if ( !pEnumerator )
+                return E_OUTOFMEMORY;
+
+            wxDirect2DFontKey key = *static_cast<wxDirect2DFontKey const*>(pCollectionKey);
+            if ( key != m_key )
+                return E_INVALIDARG;
+
+            HRESULT hr = pEnumerator->Initialize(m_fontList);
```

Comment:
Is there any reason to separate `Initialize()` from ctor? I'd pass `m_fontList` directory to the ctor and check that it's not empty here, IMHO it would slightly simplify the code.